src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -12,37 +12,66 @@
 namespace Symfony\Component\Serializer\Exception;
 
 /**
- * @deprecated since Symfony 6.3, use {@see MissingConstructorArgumentException} instead
- *
  * @author Maxime VEBER <maxime.veber@nekland.fr>
  */
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
-     * @var string[]
+     * @var string|null
+     */
+    private $class;
+
+    /**
+     * @var MissingConstructorArgumentException[]
      */
-    private $missingArguments;
+    private $missingConstructorArgumentExceptions = [];
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    /**
+     * @param MissingConstructorArgumentException[] $missingArguments
+     * @param ?class-string $class
+     */
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [], string $class = null)
     {
-        if (!$this instanceof MissingConstructorArgumentException) {
-            trigger_deprecation('symfony/serializer', '6.3', 'The "%s" class is deprecated, use "%s" instead.', __CLASS__, MissingConstructorArgumentException::class);
+        $this->missingConstructorArgumentExceptions = $missingArguments;
+        $classes = array_unique(array_map(fn (MissingConstructorArgumentException $missingConstructorArgumentException) => $missingConstructorArgumentException->getClass(), $missingArguments));
+        if (count($classes) > 1) {
+            throw new \InvalidArgumentException(sprintf(
+                'All instances of "%s" must concern the same class. Classes given : "%s".',
+                MissingConstructorArgumentException::class,
+                implode('", "', $classes)
+            ));
         }
-
-        $this->missingArguments = $missingArguments;
+        if ($class !== null && $class !== $classes[0]) {
```

Comment:
It'll fail if `$classes` is `[]` (if for example the given `$missingArguments` is empty)
You can maybe replace `$classes[0]` by `($classes[0] ?? null)`.

Reply:
Nice catch.